### DISCUSSION TITLE
Is floating button supported?

### DISCUSSION THREAD

**Discussion by stefanofornari** [palexdev/MaterialFX#219] [Archived]
Hello, this is truly a great project! Thanks for sharing this gem with the community!

Are floating buttons supported currently? Is not, is there an easy way to implement one?

I am referring to something like:

![image](https://user-images.githubusercontent.com/43138/174730948-8b8286c6-8a69-445c-9e79-73cd89a73ec5.png)

**Comment by palexdev** [comment]
Yes, take a look at the javadocs for MFXButton
Also the floating effect can be applied to all nodes thanks to MFXDepthManager

**Reply by stefanofornari** [reply]
Actually, I could not find any reference to floating buttons in the javadoc (https://javadoc.io/doc/io.github.palexdev/materialfx/latest/MaterialFX/io/github/palexdev/materialfx/effects/MFXDepthManager.html or MFXButton). Shall I look elsewhere?

**Reply by palexdev** [reply]
There is no such thing as "floating"
It is an illusion, depth is added to nodes though a DropShadow effect which makes it look like the nodes are "floating"

MFXDepthManager allows to easily apply the effect, but for MFXButtons it's recommended to use the dedicated properties which are ButtonType and DepthLevel

**Reply by stefanofornari** [reply]
Ciao Alessandro, 
with floating button I mean that primary button on the right bottom of the window and that stays on top of any widgets behind it (like the + button in the picture above). I understand it is quite standard in Material apps (but I am not an expert),I guess at the moment is not directly provided by MaterialFX and it should be done from scratch.

**Reply by palexdev** [reply]
Maybe you can achieve it with MFXButton
After all you just have to set the text to "+" and style it
As for the "stay on top" part that's standard JavaFX, children added last will always be in front of any other child
Or you can set the view order

**Reply by stefanofornari** [reply]
Yeah that's true. How can I make it round? (I know this may be a naive question, but I am a beginner with javafx)

**Reply by palexdev** [reply]
That's simple math
You need to set both width and height at the same value then set the background radius and border radius to high values (like 30, or in percentage 100%)

Since you are a beginner maybe before using MaterialFX you should follow some tutorials about the basics of JavaFX
